controller/auth.js

```diff
@@ -1,643 +1,904 @@
+  async.waterfall(array,cb);
+};
+var encode = function(data) {
+  return jwt.sign(data,tokenConfig.secret);
+};
+
+var decode = function(data,cb) {
+  jwt.verify(data,tokenConfig.secret,{},function(err, decoded_token) {
+    if (err) {
+      console.log("Error from JWT.verify", err.name);
+      console.log("Error from JWT.verify", err.message);
+    }
+    cb(decoded_token);
+  });
+};
+
+var hasExpired = function(created) {
+  var now = new Date();
+  var diff = (now.getTime() - created);
+  return diff > tokenConfig.ttl;
+};
+
+
+var invalidateUserToken = function(req,res,next,cb) {
+  var token = req.params.token;
+  User.findOne({"attributes.reset_token": token}, function(err, usr) {
+    if(err) {
+      console.log('error finding the user for invalidate token fn');
+      cb(err,null);
+    }
+    usr.attributes.reset_token = "";
+    usr.attributes.reset_token_expires_millis = 0;
+    cb(false, req,res,next,usr);
+  });
+};
+
+var resetPassword = function(req,res,next,user,cb) {
+  var err;
+  if (req.body.password) {
+    if (req.body.password !== req.body.passwordconfirm) {
+      err = new Error('Passwords do not match');
+      req.flash('error',err);
+      cb(err,null);
+    }
+    else if (req.body.password === undefined || req.body.password.length < 8) {
+      err = new Error('Password is too short');
+      req.flash('error',err);
+      cb(err,null);
+    }
+    else {
+      var salt = bcrypt.genSaltSync(10),
+      hash = bcrypt.hashSync(req.body.password, salt);
+      user.password = hash;
+      cb(null,user);
+    }
+  }
+}
+
+function saveUser(user,cb) {
+  user.markModified('attributes');
+  user.markModified('password');
+  user.save(function(err,usr) {
+    if (err) {
+      cb(err,null) 
+    }
+    cb(null,usr);
+  }) 
+}
+
+
+var getUser = function(req,res,next,cb) {
+  User.findOne(req.body.email, function(err, user) {
+    if (err) {
+      cb(err, null);
+    } else if (user) {
+      cb(false, user);
+    } else {
+      req.flash('error','No user with that email found!')
+      cb(new Error('No user with that email found.'), null);
+    }
+  });
+};
+
+var generateToken = function(user,cb) {
+  //Generate reset token and URL link; also, create expiry for reset token
+  //make sure attributes exists || create it via merge
+  var now = new Date();
+  var expires = new Date(now.getTime() + (tokenConfig.resetTokenExpiresMinutes * 60 * 1000)).getTime();
+  user.attributes = {};
+  user.attributes.reset_token = encode(user.email);
+  user.attributes.reset_token_expires_millis = expires;
+  user.markModified('attributes');
+  user.save(function(err){
+    if (err) {
+      cb(err,null);
+    }
+    cb(null,user);
+  });
+};
+
+// create a func for the mail options
+
+var emailConfig = function(user,cb) {
+  var options = {};
+  options.mailtransport = emailtransport;
+  options.subject = "You forgot your password";
+  options.user = user;
+  options.replyTo = 'ecasilla@icloud.com';
```

Comment:
lets send to user
and from:  appSettings.adminnotificationemail;

appSettings is from line 556:

var controller = function (resources) {
	logger = resources.logger;
	mongoose = resources.mongoose;
	appSettings = resources.settings;